packages/rocketchat-lib/server/methods/joinRoom.js

```diff
@@ -74,17 +74,19 @@ Meteor.methods({
 		message.attachments[0].fields[0].responder = responder;
 		return RocketChat.models.Messages.setMessageAttachments(message._id, message.attachments);
 	},
-	getJoinRoomStatus(roomId, user) {
-
+	getJoinRoomStatus(roomName) {
+		check(roomName, String);
+		const room = RocketChat.models.Rooms.findOneByName(roomName);
+		return RocketChat.models.Messages.findLatestJoinRequestByRoom('join-room-request', room._id, Meteor.user().username).fetch();
```

Comment:
`join-room-request` should not be a parameter of this method.
Either the find method should be named `findLatestByTypeAndRoomId` or this selector should be hard-coded inside the method.

Reply:
renamed as `findLatestJoinRequestByTypeAndRoomAndUser`